src/mqt/bench/benchmarks/shor.py

```diff
@@ -32,3 +52,222 @@ def get_instance(choice: str) -> list[int]:
         "xlarge": [201209, 4],  # 74 qubits
     }
     return instances[choice]
+
+
+class Shor:
+    def __init__(self, quantum_instance: QuantumInstance | Backend | None = None) -> None:
+        """
+        Args:
+            quantum_instance: Quantum Instance or Backend
+
+        """
+        self._quantum_instance = None
+        if quantum_instance:
+            self.quantum_instance = quantum_instance
+
+    @property
+    def quantum_instance(self) -> QuantumInstance | None:
+        """Returns quantum instance."""
+        return self._quantum_instance
+
+    @quantum_instance.setter
+    def quantum_instance(self, quantum_instance: QuantumInstance | Backend) -> None:
+        """Sets quantum instance."""
+        if isinstance(quantum_instance, Backend):
+            quantum_instance = QuantumInstance(quantum_instance)
+        self._quantum_instance = quantum_instance
```

Comment:
I think this can be safely removed. In general it would be good to clean this file up a little and minimize it. Although, I totaly get, that you probably don't want to do this as part of this PR.
```suggestion
```